package-res/resources/web/common-ui-require-js-cfg.js

```diff
@@ -63,6 +64,14 @@
   requirePaths["pentaho/_globalContextVars"] = basePath + "/pentaho/_globalContextVars";
   requirePaths["pentaho/contextVars"] = requirePaths["pentaho/_globalContextVars"];
 
+  // Singleton interface, virtual module ids
+  //requireMap["*"]["pentaho/spec/IContext"] = "pentaho/service!pentaho.spec.IContext?single";
+  requireMap["*"]["pentaho/config/IService"] = "pentaho/service!pentaho.config.IService?single";
```

Comment:
Like I told you, the module ID `pentaho/config/IService` seems strange. I like the map approach (@graimundo doesn't), but only if we could find a better nomenclature for these _default instances_.